browser/views.py

```diff
@@ -681,27 +705,53 @@ def edit_investigator_record(request,person_id,research_id):
     person = get_object_or_404(Person, pk=person_id)
     research = get_object_or_404(Investigator, pk=research_id)
     template = "browser/investigator.html"
+    institution_id = None
+    if research.institution:
+        institution_id = research.institution.id
 
     # Retreive investigator data already stored in database and display to the user
     form = InvestigatorForm(initial={'subject': research.subject, 'role': research.role,
-                                             'year': research.year})
+                                             'year': research.year, 'institution_search': research.institution,
+                                     'institution_id': institution_id})
+
+    context = {
+        'form': form,
+        'person': person,
+        'investigator_data': research,
+        'type': 'edit_investigator'
+    }
 
     if request.method == 'POST':
         form = InvestigatorForm(request.POST, instance=person)
+        context.update({
+            'form': form
+        })
         if form.is_valid():
+            if form.cleaned_data.get('institution_search'):
+                if not form.cleaned_data.get('institution_id'):
+                    # Check if the checkbox is checked
+                    if form.cleaned_data.get('create_institution'):
+                        institution = Institution.objects.create(
+                            name=form.cleaned_data.get('institution_search'),
+                            changed_by=request.user
+                        )
+                        form.cleaned_data["institution_id"] = institution.id
+                    else:
+                        messages.add_message(request, messages.ERROR,
+                                             'The above institute does not exist in the database. '
+                                             'Please check the checkbox below to create a new institute.')
+                        return render(request, template, context)
+
             research.subject = form.cleaned_data.get('subject')
             research.role = form.cleaned_data.get('role')
             research.year = form.cleaned_data.get('year')
+            if not form.cleaned_data.get('institution_search'):
+                research.institution = None
+            else:
+                research.institution = Institution.objects.get(id=form.cleaned_data.get('institution_id'))
```

Comment:
same here, make the if/else one line (ternary operator)

Reply:
actually no, just put that in the existing if block

Reply:
Put the above code into the existing if block